plugins/queue/memory/queue.go

```diff
@@ -0,0 +1,87 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package memory
+
+import (
+	"github.com/enriquebris/goconcurrentqueue"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/config"
+	"github.com/apache/skywalking-satellite/internal/pkg/log"
+	"github.com/apache/skywalking-satellite/internal/satellite/event"
+	"github.com/apache/skywalking-satellite/plugins/queue/api"
+	"github.com/apache/skywalking-satellite/protocol/gen-codes/satellite/protocol"
+)
+
+const (
+	Name = "memory-queue"
+)
+
+type Queue struct {
+	config.CommonFields
+	// config
+	EventBufferSize int `mapstructure:"event_buffer_size"` // The maximum buffer event size.
+
+	// components
+	buffer *goconcurrentqueue.FixedFIFO
```

Comment:
Forgive me don't read their documentation first, is this FIFO queue being blocked when the queue is full? If so, do they provide any policy to abandon data or something?

Reply:
yes, the queue would reject the new data.